### PR TITLE
Support a blank expression before a semicolon

### PR DESCRIPTION
**Pull request by DJMcNab**
Part of fixing #290.

Note that I have not run `gen-tests` as I don't want to conflict with #299.

### CODE REVIEW

File: crates/ra_syntax/src/grammar/expressions.rs [hunk 1]
```diff
@@ -42,6 +42,11 @@ pub(crate) fn block(p: &mut Parser) {
     }
     let m = p.start();
     p.bump();
+    // test nocontentexpr
+    // fn foo(){
+    //     ;;;some_expr();;;;{;;;};;;;Ok(())
+    // }
+    while p.eat(SEMI) {}
```
Comment (matklad) [inline crates/ra_syntax/src/grammar/expressions.rs]: Hm, can we perhaps have only one `while p.eat(SEMI)` loop? I think if you put it right before `match` it'll do the trick? Or maybe even just add `SEMI => p.bump(),` case to match itself?

**Comment by matklad** [comment]
bors delegate+

File: crates/ra_syntax/src/grammar/expressions.rs (hunk 1 above)
Comment (DJMcNab) [reply crates/ra_syntax/src/grammar/expressions.rs]: I fixed a couple of bugs, which got rid of error nodes. That could feasibly be the reason

**Comment by matklad** [comment]
bors try

**Pull request merged**